eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ThrownClientLoggerCheck.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * To throw an exception,
+ * <ol>
+ *   <li>Must do it through a 'logger.logAndThrow', rather than by directly calling 'throw exception'</li>
+ *   <li>Always call return after called 'logger.logAndThrow'.</li>
+ * </ol>
+ *
+ * Skip check if throw exception from
+ * <ol>
+ *   <li>Static method</li>
+ *   <li>Static class</li>
+ *   <li>Constructor</li>
+ * </ol>
+ */
+public class ThrownClientLoggerCheck extends AbstractCheck {
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_THROW
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.LITERAL_THROW:
+                DetailAST parentToken = token.getParent();
+                boolean throwsFromMethod = false;
+                while (parentToken != null) {
+                    int parentType = parentToken.getType();
+                    // Skip throw exception check if LITERAL_THROW from static method
+                    if (parentType == TokenTypes.METHOD_DEF) {
+                        DetailAST modifiersToken = parentToken.findFirstToken(TokenTypes.MODIFIERS);
+                        if (modifiersToken.branchContains(TokenTypes.LITERAL_STATIC)) {
+                            return;
+                        }
+                        throwsFromMethod = true;
+                    }
+                    // Skip throw exception check if LITERAL_THROW from static class
+                    if (parentType == TokenTypes.CLASS_DEF) {
+                        DetailAST modifiersToken = parentToken.findFirstToken(TokenTypes.MODIFIERS);
+                        if (modifiersToken.branchContains(TokenTypes.LITERAL_STATIC)) {
+                            return;
+                        }
+                        break;
+                    }
+                    // Skip throw exception check if LITERAL_THROW from constructor
+                    if (!throwsFromMethod && parentType == TokenTypes.CTOR_DEF) {
+                        return;
+                    }
+                    parentToken = parentToken.getParent();
+                }
+
+                log(token, "To throw an exception, must do it through a ''logger.logAndThrow'', rather than "
```

Comment:
nit: "Directly throwing an exception is disallowed.  Replace the throw statement with a call to 'logger.logAndThrow'."